test/e2e/operator_test.go

```diff
@@ -0,0 +1,279 @@
+	err = waitForOperatorToBeReady(t, ctx)
+	if err != nil {
+		t.Fatalf("ClusterOperator never became ready: %v", err)
+	}
+
+	// Ensure the VolumeSnapshot CRD is installed
+	// I wonder if this is even useful, and if it should be removed, since we check that the Operator is available?
+	_, err = ctx.ApiextensionsV1beta1Interface.CustomResourceDefinitions().Get("volumesnapshots.snapshot.storage.k8s.io", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error attempting to retrieve snapshot CRD: %v", err)
+	}
+
+	// Create the PV
+	pv := helpers.CreateFakePV(t, "pv"+strconv.Itoa(rand.Intn(1000)), driver.GetDriverInfo().Name, "pv", size)
+	pv, err = ctx.CoreV1Interface.PersistentVolumes().Create(pv)
+	if err != nil {
+		t.Fatalf("Error attempting to create PV: %v", err)
+	}
+	defer func() {
+		err = ctx.CoreV1Interface.PersistentVolumes().Delete(pv.ObjectMeta.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("Error attempting to delete PV: %v", err)
+		}
+	}()
+
+	// Create the PVC
+	pvc := helpers.CreateFakePVC("pvc"+strconv.Itoa(rand.Intn(1000)), namespace, pv.Spec.StorageClassName, pv.ObjectMeta.UID, size)
+	pvc, err = ctx.CoreV1Interface.PersistentVolumeClaims(namespace).Create(pvc)
+	if err != nil {
+		t.Fatalf("Error attempting to create PVC: %v", err)
+	}
+	defer func() {
+		err = ctx.CoreV1Interface.PersistentVolumeClaims(namespace).Delete(pvc.ObjectMeta.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("Error attempting to delete PVC: %v", err)
+		}
+	}()
+
+	// Create the VolumeSnapshotClass
+	snapshotClass := helpers.CreateFakeSnapshotClass(driver.GetDriverInfo().Name, "test-snapshotclass"+strconv.Itoa(rand.Intn(1000)))
+	snapshotClass, err = ctx.SnapshotV1beta1Interface.VolumeSnapshotClasses().Create(snapshotClass)
+	if err != nil {
+		t.Fatalf("Error attempting to create VolumeSnapshotClass: %v", err)
+	}
+	defer func() {
+		err = ctx.SnapshotV1beta1Interface.VolumeSnapshotClasses().Delete(snapshotClass.ObjectMeta.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("Error attempting to delete VolumeSnapshotClass: %v", err)
+		}
+	}()
+	t.Logf("Created snapshotClass: %v", snapshotClass.ObjectMeta.Name)
+
+	// Create the VolumeSnapshot
+	snapshot := helpers.CreateFakeSnapshot(pvc.ObjectMeta.Name, namespace, "test-snapshot"+strconv.Itoa(rand.Intn(1000)), snapshotClass.ObjectMeta.Name, pvc.ObjectMeta.UID)
+	snapshot, err = ctx.SnapshotV1beta1Interface.VolumeSnapshots(namespace).Create(snapshot)
+	if err != nil {
+		t.Fatalf("Error attempting to create VolumeSnapshot: %v", err)
+	}
+	defer func() {
+		err = ctx.SnapshotV1beta1Interface.VolumeSnapshots(namespace).Delete(snapshot.ObjectMeta.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("Error attempting to delete VolumeSnapshot: %v", err)
+		}
+	}()
+	t.Logf("Created snapshot %v", snapshot.ObjectMeta.Name)
+
+	// Search for a matching VolumeSnapshotContent
+	snapshotContent, err := searchForSnapshotContent(t, ctx, snapshot)
```

Comment:
`waitForSnapshotContent` is more idiomatic?

Reply:
Adjusted the function name.